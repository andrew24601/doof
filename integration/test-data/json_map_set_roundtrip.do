class Bag {
  tags: Map<string, string>;
  ids: Set<int>;
}

function main(): void {
  // Create a Bag with Map/Set
  let tags: Map<string, string> = {};
  tags["a"] = "1";
  tags["b"] = "2";

  let ids: Set<int> = {};
  ids.add(7);

  let bag: Bag = { tags: tags, ids: ids };

  // Deserialize from JSON string (matches normalized shape)
  let json: string = "{\"tags\":{\"a\":\"1\",\"b\":\"2\"},\"ids\":[7]}";
  let restored: Bag = Bag.fromJSON(json);

  // Expect same normalized JSON on all backends that support fromJSON for Map/Set
  println(restored);
}
