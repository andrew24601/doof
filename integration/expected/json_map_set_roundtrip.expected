{"tags":{"a":"1","b":"2"},"ids":[7]}
